src/checkers/inference/util/JaifBuilder.java

```diff
@@ -310,7 +310,12 @@ private void buildClassEntries() {
                     // TODO: this is not a feature but a workaround of a bug:
```

Comment:
Note that this is just a hacky work-around.
I think it would be better to fix all the places that incorrectly have an empty ASTPath. Instead of a silent `continue` here, we should raise an error when we still find an empty ASTPath.
Also note the duplication between the change here and the one above. Do we really need both? If so, maybe the `getConstructorRecord` method should be called in more places?

As discussed, please first add a test case to make sure we're fixing this error.